### PR TITLE
new file: src/downloadcontent.py

### PR DESCRIPTION
**Pull request by sergiomb2**
draft for #5

### CODE REVIEW

File: src/vnu_validator/__init__.py [hunk 1]
```diff
@@ -178,7 +179,8 @@ def run(self):
         """
         # t = tempfile.TemporaryDirectory()
         tname = tempfile.mkdtemp()
-        _bin_spew(join(tname, "foo.html"), _bin_slurp(self.path))
+        url, content, text_aux = downloadcontent(self.path)
```
Comment (shlomif) [inline src/vnu_validator/__init__.py:182]: Perhaps extract a `fetch_path()` method, a base class and two child classes. i'd rather not conflate https fetching w local files

File: src/vnu_validator/__init__.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from subprocess import PIPE, Popen
 
 from six.moves.urllib.parse import urlparse
+from vnu_validator.downloadcontent import downloadcontent
```
Comment (shlomif) [inline src/vnu_validator/__init__.py:15]: Please localize this import.

File: src/vnu_validator/downloadcontent.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import http.client
+import re
+import sys
+#to print stack information
+import traceback
+import time
+from sys import argv
+
+
+def downloadcontent(url, extras = {}, log = None):
```
Comment (shlomif) [inline src/vnu_validator/downloadcontent.py:16]: this is generic enough to be its own pypi dist

File: src/vnu_validator/downloadcontent.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+    if img.strip() == "":
+        typemesg = "error"
+        dbmesg = "Error img/pdf is empty ! (%s)" % (url)
+        return final_url, typemesg, dbmesg
+
+    #print final_url, img
+    return final_url, img, ""
+
+if __name__ == '__main__':
```
Comment (shlomif) [inline src/vnu_validator/downloadcontent.py:88]: all of this and its imports doesnt belong here.

File: src/vnu_validator/downloadcontent.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    log = logging.getLogger('downloadcontent')
+
+    if len(argv) < 1 or len(argv) > 2 :
+        print("Usage : %s infile [extras]" % sys.argv[0])
+        exit(1)
+    extras = {}
+    if len(argv) > 1:
+        try:
+            extras = eval (argv[2])
```
Comment (shlomif) [inline src/vnu_validator/downloadcontent.py:97]: "eval()"? god help me...

**Comment by shlomif** [comment]
@sergiomb2 : see my comments.

**Comment by sergiomb2** [comment]
> @sergiomb2 : see my comments.

where ?

**Comment by shlomif** [comment]
> > @sergiomb2 : see my comments.
> 
> where ?

my code review comments on https://github.com/shlomif/python-vnu_validator/pull/6/files . Thanks!

**Comment by sergiomb2** [comment]
image

I don't see it !

**Review by shlomif — CHANGES_REQUESTED** [review]
See my comments,

**Comment by shlomif** [comment]
@sergiomb2 : see my comments.

> image
> 
> I don't see it !

Hope it is ok now. Stupid github UI: https://www.shlomifish.org/philosophy/books-recommends/#design_of_everyday_things

**Comment by shlomif** [comment]
@sergiomb2 : hi! it should be ok now. Did you read my comments? Please confirm.

**Comment by sergiomb2** [comment]
yes I read your comments , this weekend (tomorrow and after tomorrow) I will improve my commit ,
I will replace eval with  json.loads(x) , I think

**Comment by shlomif** [comment]
> yes I read your comments , this weekend (tomorrow and after tomorrow) I will improve my commit , I will replace eval with json.loads(x) , I think

Gotcha. I think the whole `main()` section can be removed altogether.

**Comment by shlomif** [comment]
@sergiomb2 any news?

**Comment by shlomif** [comment]
Rejecting for now, to avoid clutter.

**Pull request closed without merging**